src/services/layout-service.ts

```diff
@@ -22,16 +25,22 @@ const createLayoutService = (
   const dimensionInfoIndexMap: Map<VisibleDimensionInfo, number> = new Map(
     qEffectiveInterColumnSortOrder.map((index) => [qDimensionInfo[index] ?? PSEUDO_DIMENSION_INDEX, index]),
   );
+  const visibleMeasureInfo = qMeasureInfo.filter((info) => info.qError?.qErrorCode !== HIDDEN_ERROR_CODE);
 
   return {
     layout,
     getNullValueText: () => nullValueRepresentation?.text ?? "-",
-    getMeasureInfoIndexFromCellIndex: (index: number) => {
+    getMeasureInfoIndexFromCellIndex: (index: number, getVisibleIndex = false) => {
       if (hasPseudoDimOnLeft) {
         return 0;
       }
 
-      return index % qMeasureInfo.length;
+      const visibleMeasureInfoIndex = index % visibleMeasureInfo.length;
+      if (getVisibleIndex) {
+        return visibleMeasureInfoIndex;
+      }
+
+      return qMeasureInfo.findIndex((info) => info === visibleMeasureInfo[visibleMeasureInfoIndex]);
```

Comment:
I think this part could be true for any object: 
```tsx
info === visibleMeasureInfo[visibleMeasureInfoIndex]
```

because `visibleMeasureInfo` is resulted from `.filter(...)` which will be a new set of arrays, so i think even references of those measure objects will not match together? 

probably an alternate solution could be checking the `cId` or `qLibraryId` or something else within the measure obj.

Reply:
It will match since the object reference inside `qMeasureInfo[]` and `visibleMeasureInfo[]` are still the same.

It should work like this:
```js
const m0 = { visible: false };
const m1 = { visible: true };
const m2 = { visible: true };

const qMeasureInfo  = [m0, m1, m2];
const visibleMeasureInfo = qMeasureInfo.filter(info => info.visible);

return qMeasureInfo.findIndex(info => visibleMeasureInfo[1]); // Returns index 2 for "m2"
```

Reply:
if you run that code in console -> the findIndex will give you 0 matching the first object in it's search flow through `qMeasureInfo`, and that is the problem that was mentioning

Reply:
Will take a look

Reply:
It was a typo in the example. Should work with this fix:

```js
const m0 = { visible: false };
const m1 = { visible: true };
const m2 = { visible: true };

const qMeasureInfo  = [m0, m1, m2];
const visibleMeasureInfo = qMeasureInfo.filter(info => info.visible);

return qMeasureInfo.findIndex(info => info === visibleMeasureInfo[1]); // Returns index 2 for "m2"
```